modules/auxiliary/scanner/ssl/ssl_version.rb

```diff
@@ -0,0 +1,463 @@
+            host: ip,
+            name: vhostn
+          )
+        end
+
+      end
+    else
+      print_status("\tNo certificate subject or common name found.")
+    end
+  end
+
+  def check_vulnerabilities(ip, ssl_version, ssl_cipher, cert)
+    # POODLE
+    if ssl_version == 'SSLv3'
+      print_good('Accepts SSLv3, vulnerable to POODLE')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv3 is available. Vulenrable to POODLE, CVE-2014-3566.",
+        refs: 'CVE-2014-3566',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # DROWN
+    if ssl_version == 'SSLv2'
+      print_good('Accepts SSLv2, vulnerable to DROWN')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv2 is available. Vulenrable to DROWN, CVE-2016-0800.",
+        refs: 'CVE-2016-0800',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # BEAST
+    if ((ssl_version == 'SSLv3') || (ssl_version == 'TLSv1.0')) && ssl_cipher.include?('CBC')
+      print_good('Accepts SSLv3/TLSv1 and a CBC cipher, vulnerable to BEAST')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed SSLv3/TLSv1 and a CBC cipher. Vulenrable to BEAST, CVE-2011-3389.",
+        refs: 'CVE-2011-3389',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # RC4 ciphers
+    if ssl_cipher.upcase.include?('RC4')
+      print_good('Accepts RC4 cipher.')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed RC4 cipher. CVE-2013-2566.",
+        refs: 'CVE-2013-2566',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # export ciphers
+    if ssl_cipher.upcase.include?('EXPORT')
+      print_good('Accepts EXPORT based cipher.')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed EXPORT based cipher.",
+        refs: references,
+        exploited_at: Time.now.utc
+      )
+    end
+    # LOGJAM
+    if ssl_cipher.upcase.include?('DHE_EXPORT')
+      print_good('Accepts DHE_EXPORT based cipher.')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed DHE_EXPORT based cipher. Vulnerable to LOGJAM, CVE-2015-4000",
+        refs: 'CVE-2015-4000',
+        exploited_at: Time.now.utc
+      )
+    end
+
+    # Null ciphers
+    if ssl_cipher.upcase.include? 'NULL'
+      print_good('Accepts Null cipher')
+      report_vuln(
+        host: ip,
+        port: rport,
+        proto: 'tcp',
+        name: name,
+        info: "Module #{fullname} confirmed Null cipher.",
+        refs: 'CVE-2011-3389',
```

Comment:
This CVE is the same as BEAST? Are you sure that is right, as this looks like a separate vuln.